### PR TITLE
feat: expose schema via control API

### LINKED ISSUE #3022: feat: support to export the JSON schema of APISIX via `CLI`
discuss mail:
https://lists.apache.org/thread.html/rb1c19fe9b6e51ede3b5aead122e2bcd4de6249f42b971eabfb8a79ac%40%3Cdev.apisix.apache.org%3E

Because the dashboard project requires schema verification for plug-ins,
upstreams, etc., and the definition of the schema is written in the Lua
code of Apache APISIX. Therefore, we need a tool to generate a JSON schema
file based on the Lua code of Apache APISIX so that it can be used by the
Go and front-end code of the dashboard project.

If the developer want to sync the JSON schema, he/she should follow this step:

1. export JSON schema file: call `apisix export-jsonschema`, we will get the file `schema.json`
2. copy the file `schema.json` from apisix to the dashboard folder
3. start the dashboard

those steps are much safer. and the dashboard does not need any update.

### PR DESCRIPTION
**Pull request by spacewander**
Close #3022

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/control/v1.lua [hunk 1]
```diff
@@ -14,11 +14,33 @@
...
-    return 200, {}
+    local schema = {
+        main = {
+            global_rule = core.schema.global_rule,
+            route = core.schema.route,
+            service = core.schema.service,
+            ssl = core.schema.ssl,
+            stream_route = core.schema.stream_route,
+            upstream = core.schema.upstream,
+        },
```
Comment (nic-chen) [inline apisix/control/v1.lua:33]: how about could use `core.schema` as main ?

There will be redundant data, but no data will not be missed, and there is no need to modify the code here after adding a new schema.

Comment (spacewander) [reply apisix/control/v1.lua:33]: There is some methods in `core.schema` can't be json encoded, like `core.schema.check`.

Comment (nic-chen) [reply apisix/control/v1.lua:33]: ok

**Review by nic-chen — APPROVED** [review]

File: apisix/control/v1.lua (hunk 1 above)
Comment (nic-chen) [reply apisix/control/v1.lua:33]: lack of consumer

**Review by nic-chen — CHANGES_REQUESTED** [review]

File: apisix/control/v1.lua (hunk 1 above)
Comment (spacewander) [reply apisix/control/v1.lua:36]: @nic-chen 
Solved.

**Review by nic-chen — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Pull request merged**